homeassistant/components/adguard/sensor.py

```diff
@@ -103,17 +104,17 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up AdGuard Home sensor based on a config entry."""
-    adguard = hass.data[DOMAIN][entry.entry_id][DATA_ADGUARD_CLIENT]
+    data = hass.data[ADGUARD_HASS_KEY][entry.entry_id]
```

Comment:
The downside for me is that I can't just read the platform code anymore and know what the type is of `data`. As the type is different for each integration it's also not something you'll likely learn by heart compared to the core or helper interface functions. I'll have to jump around and check the type when reading the integration code.

Reply:
To be clear, this is what I'm comparing:
```py
data: AdGuardData = hass.data[DOMAIN][entry.entry_id]
```

Reply:
That's true, although I would argue it's manageable with good code completion. Also consider that the type (and any attributes) would get type checked and you're less likely to make an error as with type attributes on each individual platform. (Who says it's the correct type that's being assigned in `sensor.py`?)